src/lib/y2partitioner/widgets/blk_devices_table.rb

```diff
@@ -263,66 +250,134 @@ def type_icon(device)
+
+        if fs && fs.multidevice?
+          btrfs_multidevice_type_label(fs)
+        elsif fs
+          [fs.type.to_human_string, label].join(" ")
+        else
+          label
+        end
+      end
+
+      # Label for the device type (e.g., LVM, RAID, Partition, etc)
+      #
+      # @param device [Y2Storage::BlkDevice]
+      # @return [String]
+      def device_label(device)
+        if device.is?(:lvm_vg)
+          device_label_for(:lvm_vg)
         elsif device.is?(:lvm_lv)
           lvm_lv_type_label(device)
+        elsif device.lvm_pv
+          lvm_pv_type_label(device.lvm_pv)
+        elsif device.bcache
+          part_of_label(device.bcache)
+        elsif device.in_bcache_cset
+          bcache_cset_label
+        elsif device.md
+          part_of_label(device.md)
+        elsif device.is?(:partition)
+          partition_type_label(device)
         else
-          blk_device_type_label(device)
+          default_type_label(device)
         end
       end
 
-      # Default type label for the device
+      # Type label when the device is a LVM logical volume
       #
-      # @see DEVICE_LABELS
+      # @note Different label is shown depending on the logical volume type
+      #   (i.e., normal, thin pool or thin volume).
       #
-      # @param device [Y2Storage::BlkDevice]
+      # @param device [Y2Storage::LvmLv]
       # @return [String]
-      def default_type_label(device)
-        type = DEVICE_LABELS.keys.find { |k| device.is?(k) }
-        return "" if type.nil?
+      def lvm_lv_type_label(device)
+        return default_type_label(device) if device.lv_type.is?(:normal)
 
-        _(DEVICE_LABELS[type])
+        type = device.lv_type.to_sym
+        device_label_for(type) || ""
+      end
+
+      # Type label when the device is a LVM physical volume
+      #
+      # @param device [Y2Storage::LvmPv]
+      # @return [String]
+      def lvm_pv_type_label(device)
+        vg = device.lvm_vg
+
+        return _("Orphan PV") if vg.nil?
+
+        # TRANSLATORS: %s is the volume group name. E.g., "vg0"
+        format(_("PV of %s"), vg.basename)
+      end
+
+      # Type label when the device belongs to a multidevice BtrFS filesystem
+      #
+      # @param fs [Y2Storage::Filesystem]
+      # @return [String]
+      def btrfs_multidevice_type_label(fs)
+        # TRANSLATORS: %s is a mount path. E.g., "/", or "/home"
+        format(_("Part of BtrFS %s"), fs.mount_path)
+      end
+
+      # Type label when the device is used as BcacheCSet
+      #
+      # @return [String]
+      def bcache_cset_label
+        # TRANSLATORS: an special type of device
+        _("Bcache cache")
       end
 
       # Type label when the device is a partition
       #
       # @param device [Y2Storage::Partition]
       # @return [String]
       def partition_type_label(device)
-        device.id.to_human_string
+        return _("Partition") if device.filesystem
+
+        format(_("%s Partition"), device.id.to_human_string)
```

Comment:
Actually, I think that in this case we should return only the `device.id`. Doing that, we'll avoid having the same type when the partition has a filesystem and when not. I.e.,

* `Swap Partition` when has a Swap filesystem
* `Swap` when has no filesystem but Swap id.

instead of having `Swap Partition` for both cases.

What do you think?